cartography/intel/azure/function_app.py

```diff
@@ -0,0 +1,707 @@
+            function_apps_webjobs_list = function_apps_webjobs_list + list(
+                map(
+                    lambda x: x.as_dict(),
+                    client.web_apps.list_web_jobs(function['resource_group'],
+                                                  function['name'])))
+
+        for webjob in function_apps_webjobs_list:
+            x = webjob['id'].split('/')
+            webjob['resource_group'] = x[x.index('resourceGroups') + 1]
+            webjob['function_app_id'] = webjob['id'][:webjob['id'].
+                                                     index("/snapshots")]
+        return function_apps_webjobs_list
+
+    except HttpResponseError as e:
+        logger.warning(f"Error while retrieving function apps webjobs - {e}")
+        return []
+
+
+def _load_function_apps_webjobs_tx(
+    tx: neo4j.Transaction,
+    function_apps_webjobs_list: List[Dict],
+    update_tag: int,
+) -> None:
+    ingest_function_apps_webjob = """
+    UNWIND {function_apps_webjobs_list} as function_webjob
+    MERGE (f:AzureFunctionAppWebjob{id: function_webjob.id})
+    ON CREATE SET f.firstseen = timestamp(),
+    f.type = function_webjob.type
+    SET f.name = function_webjob.name,
+    f.lastupdated = {azure_update_tag},
+    f.resource_group_name=function_webjob.resource_group
+    WITH f, function_webjob
+    MATCH (s:AzureFunctionApp{id: function_webjob.function_app_id})
+    MERGE (s)-[r:CONTAIN]->(f)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {azure_update_tag}
+    """
+
+    tx.run(
+        ingest_function_apps_webjob,
+        function_apps_webjobs_list=function_apps_webjobs_list,
+        azure_update_tag=update_tag,
+    )
+
+
+def cleanup_function_apps_webjobs(neo4j_session: neo4j.Session,
+                                  common_job_parameters: Dict) -> None:
+    run_cleanup_job('azure_import_function_apps_webjobs_cleanup.json',
+                    neo4j_session, common_job_parameters)
+
+
+def sync_function_apps_webjobs(
+    neo4j_session: neo4j.Session,
+    credentials: Credentials,
+    subscription_id: str,
+    update_tag: int,
+    common_job_parameters: Dict,
+) -> None:
+    function_apps_webjobs_list = get_function_apps_webjobs_list(
+        credentials, subscription_id)
+    load_function_apps_webjobs(neo4j_session, function_apps_webjobs_list,
+                               update_tag)
+    cleanup_function_apps_webjobs(neo4j_session, common_job_parameters)
+
+
+@timeit
+def sync(
+    neo4j_session: neo4j.Session,
+    credentials: Credentials,
+    subscription_id: str,
+    update_tag: int,
+    common_job_parameters: Dict,
+) -> None:
+    logger.info("Syncing function apps for subscription '%s'.",
+                subscription_id)
+
+    sync_function_apps(neo4j_session, credentials, subscription_id, update_tag,
+                       common_job_parameters)
+    sync_function_apps_conf(neo4j_session, credentials, subscription_id,
```

Comment:
all of these sync statements need function_apps. so, we should pass function_apps read as part of sync_function_apps to all the following sync_.... functions